test/commands/platform-dependency-version.test.ts

```diff
@@ -0,0 +1,145 @@
+import {expect, test} from '@oclif/test'
+import {readFile} from 'node:fs/promises'
+
+describe('platform-dependency-version', () => {
+  describe('checks dependencies versions for a given WP version', () => {
+    test
+    .stdout()
+    .command(['platform-dependency-version', '--wpVersion=6.0.3', 'react', '@wordpress/data', 'foo'])
+    .it('renders it in a table', ctx => {
+      // Header contains the WP version.
+      expect(ctx.stdout).to.have.string('WordPress 6.0.3')
+
+      // Prints packages with their versions.
+      expect(ctx.stdout).to.match(/@wordpress\/data\s*6\.6\.1/ms)
+      expect(ctx.stdout).to.match(/react\s*17\.0\.2/ms)
+      expect(ctx.stdout).to.match(/foo\s*$/ms) // No version for unmatched package
+    })
+
+    test
+    .stdout()
+    .command(['platform-dependency-version', '--wpVersion=6.0.3', '@wordpress/components', 'baz', '--dependenciesJSON=test/mocks/.externalized.json'])
+    .it('with --dependenciesJSON takes dependencies in from JSON file, ignoring the ones from arguments', ctx => {
+      // Header contains the WP version.
+      expect(ctx.stdout).to.have.string('WordPress 6.0.3')
+
+      // Prints packages with their versions.
+      expect(ctx.stdout).to.match(/@wordpress\/data\s*6\.6\.1/ms)
+      expect(ctx.stdout).to.match(/react\s*17\.0\.2/ms)
+      expect(ctx.stdout).to.match(/foo\s*$/ms) // No version for unmatched package
+
+      // Does not print packages from arguments.
+      expect(ctx.stdout).to.not.match(/@wordpress\/components/ms)
+      expect(ctx.stdout).to.not.match(/baz/)
+    })
+
+    test
+    .stdout()
+    .command(['platform-dependency-version', '--wpVersion=6.0.3', '--dependenciesJSON=test/mocks/.externalized.json', '--json'])
+    .it('with `--json` returns them in JSON format', ctx => {
+      const result = ctx.returned as string[]
+      expect(result).to.deep.equal(
+        [
+          {name: 'react', wpVersion: '17.0.2', wcVersion: null, localVersion: null},
+          {name: '@wordpress/data', wpVersion: '6.6.1', wcVersion: null, localVersion: null},
+          {name: 'foo', wpVersion: null, wcVersion: null, localVersion: null},
+          {name: '@woocommerce/components', wpVersion: null, wcVersion: null, localVersion: null},
+          {name: '@woocommerce/settings', wpVersion: null, wcVersion: null, localVersion: null},
+        ],
+      )
+    })
+  })
+
+  describe('if there is a package.json in working directory, it checks the dependency versions there', async () => {
```

Comment:
All tests under this `describe` block were not performed. It seems to be mainly affected by the `async` syntax.

Reply:
Addressed in https://github.com/woocommerce/dewped/pull/5/commits/b9b142c9b694ea8e86fd7b5dbd9f5a1522122d20